checkov/common/output/record.py

```diff
@@ -152,6 +154,18 @@ def to_string(self, compact: bool = False, use_bc_ids: bool = False) -> str:
         code_lines = ""
         if self.code_block:
             code_lines = "\n{}\n".format("".join([self._code_line_string(self.code_block, not(ANSI_COLORS_DISABLED))]))
+
+        detail = ""
+        if len(self.details):
+            detail_buffer = ""
+
+            detail_buffer += colored(f"\tDetails: {self.details[0]}\n", "blue")
+
+            for t in self.details[1:]:
+                detail_buffer += colored(f"\t\t{t}\n", "blue")
+
+            detail = detail_buffer
```

Comment:
```suggestion
        if self.details:
            detail_buffer = [colored(f"\tDetails: {self.details[0]}\n", "blue")]

            for t in self.details[1:]:
                detail_buffer.append(colored(f"\t\t{t}\n", "blue"))

            detail = "".join(detail_buffer)
```
using string concatenation with a `+` results in Python creating multiple strings, therefore creating a list of strings and then joining them is most of the time faster and uses less memory.